Feature/26 registration page

lily-law opened the pull request:
This PR completes the prototype signup page.
It also:
- Adds an option to include a errorMessageReturner prop function on Input component. This disables displaying the message within the component and returns the message to the specified function instead
- Fixes a mistake take would give the password validator the opposite result

lily-law committed:
Fix inverted valiation assertion

lily-law committed:
Make name validation and error message utils

lily-law committed:
Add optional errorMessageReturner to Input

lily-law committed:
Correct Login error assertion

lily-law committed:
Add Register component
- Style Register compoent

lily-law committed:
Update and style Signup page

lily-law committed:
Update Register component styles

davideastmond reviewed (approved):
Nice!!

davideastmond commented on app/src/utils/passwordValidations.tsx:
```diff
@@ -2,7 +2,7 @@ const isPasswordTooShort = (password: string) => {
 };
 const doesPasswordHaveMixedCase = (password: string) => {
-  return !password.match(/[A-Z]/) || !password.match(/[a-z]/);
+  return password.match(/[A-Z]/) || password.match(/[a-z]/);
```
Are we testing here if the password has at least one upper and one lowercase letter?
If so, I was getting unexpected results when testing. 
If you are trying to test this, this is what I came up with. There might be a more succinct way of doing it?
```suggestion
  const lowerCaseExpression = /[a-z]/;
  const upperCaseExpression = /[A-Z]/;
  return lowerCaseExpression.test(password) && upperCaseExpression.test(password);
```

lily-law replied:
Nice catch David, I completely missed changing from or to and when I flipped the match statements. Your solution is much clearer here 🙏

lily-law committed:
Update app/src/utils/passwordValidations.tsx

Co-authored-by: David Eastmond <davebuck1981@gmail.com>

lily-law committed:
Add REACT_APP_API_PORT to .env
- Update Login to use REACT_APP_API_PORT env
- Update Register to use REACT_APP_API_PORT env

lily-law committed:
Fix console warnings

lily-law committed:
Fix typo in sample.env

lily-law committed:
Update Login and Register so submit checks errors
